modules/pointcloudio/src/coders_obj.cpp

```diff
@@ -0,0 +1,80 @@
+#include "coders_obj.hpp"
+#include <iostream>
+#include <opencv2/core/utils/logger.hpp>
+
+namespace cv
+{
+namespace pc
+{
+
+void ObjDecoder::readData(std::vector<Point3f> &points, std::vector<Point3f> &normals)
+{
+    std::vector<Point3f> temp_vertices;
+    std::vector<Point3f> temp_normals;
+
+    FILE *file = fopen(m_filename.c_str(), "r");
+    if (file == NULL)
+    {
+        CV_Error(Error::StsError, "Impossible to open the file !\n");
+    }
+
+    char lineHeader[128];
+
+    while (1)
+    {
+
+        int res = fscanf(file, "%s", lineHeader);
+        if (res == EOF)
+            break;
+
+        if (strcmp(lineHeader, "v") == 0)
+        {
+            Point3f vertex;
+            res = fscanf(file, "%f %f %f\n", &vertex.x, &vertex.y, &vertex.z);
+            temp_vertices.push_back(vertex);
+        }
+        else if (strcmp(lineHeader, "vn") == 0)
+        {
+            Point3f normal;
+            res = fscanf(file, "%f %f %f\n", &normal.x, &normal.y, &normal.z);
+            temp_normals.push_back(normal);
+        }
+    }
+
+    fclose(file);
+
+    points = temp_vertices;
```

Comment:
unnecessary copy. use originals vectors. you can clean it before push_back by `.clear()`